api/src/auth/apiRequest/apiRequestErrors.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * List of errors that could occur during the authentication handshake between client extension and Azure Resources host extension.
+ */
+export const AzureResourcesApiRequestErrors = {
```

Comment:
I think there should be a type here to enforce the shape on the errors.

Reply:
This suggestion led me to a really cool discovery.  I wanted to keep the code values not typed as strings but as the actual string values, so I was using `as const` here.  But it turns out you can use `as const` AND `satisfies` in the same line, which is pretty cool and lets me accomplish what I wanted originally while enforcing the shape like you suggested.  Maybe not as interesting if you already knew that, but I thought it was pretty neat 😆